src/Evolution/Systems/GeneralizedHarmonic/ConstraintDamping/TimeDependentTripleGaussian.hpp

```diff
@@ -0,0 +1,220 @@
+ * `double` or `DataVector`, `Fr` is a frame (e.g. `Frame::Inertial`), and
+ * `VolumeDim` is the dimension of the spatial volume. The Gaussian widths
+ * \f$w_\alpha\f$ are scaled by the inverse of the value of a scalar
+ * domain::FunctionsOfTime::FunctionOfTime \f$f(t)\f$ named
+ * `FunctionOfTimeForScaling`: \f$w_\alpha(t) = w_\alpha / f(t)\f$.
+ */
+template <size_t VolumeDim, typename Fr>
+class TimeDependentTripleGaussian : public DampingFunction<VolumeDim, Fr> {
+ public:
+  struct Constant {
+    using type = double;
+    static constexpr Options::String help = {"The constant."};
+  };
+
+  struct Amplitude1 {
+    using type = double;
+    static constexpr Options::String help = {"The amplitudes of Gaussian 1."};
+  };
+
+  struct Width1 {
+    using type = double;
+    static constexpr Options::String help = {
+        "The unscaled widths of Gaussian 1."};
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  struct Center1 {
+    using type = std::array<double, VolumeDim>;
+    static constexpr Options::String help = {"The centers of Gaussian 1."};
+  };
+
+  struct Amplitude2 {
+    using type = double;
+    static constexpr Options::String help = {"The amplitudes of Gaussian 2."};
+  };
+
+  struct Width2 {
+    using type = double;
+    static constexpr Options::String help = {
+        "The unscaled widths of Gaussian 2."};
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  struct Center2 {
+    using type = std::array<double, VolumeDim>;
+    static constexpr Options::String help = {"The centers of Gaussian 2."};
+  };
+
+  struct Amplitude3 {
+    using type = double;
+    static constexpr Options::String help = {"The amplitudes of Gaussian 3."};
+  };
+
+  struct Width3 {
+    using type = double;
+    static constexpr Options::String help = {
+        "The unscaled widths of Gaussian 3."};
+    static type lower_bound() noexcept { return 0.; }
+  };
+
+  struct Center3 {
+    using type = std::array<double, VolumeDim>;
+    static constexpr Options::String help = {"The centers of Gaussian 3."};
+  };
+
+  struct FunctionOfTimeForScaling {
+    using type = std::string;
+    static constexpr Options::String help = {"The name of the FunctionOfTime."};
+  };
+
+  using options = tmpl::list<Constant, Amplitude1, Width1, Center1, Amplitude2,
+                             Width2, Center2, Amplitude3, Width3, Center3,
+                             FunctionOfTimeForScaling>;
+
+  static constexpr Options::String help = {
+      "Computes a sum of a constant and 3 Gaussians (each with its own "
+      "amplitude, width, and coordinate center), with the Gaussian widths "
+      "scaled by the inverse of a FunctionOfTime."};
+
+  /// \cond
+  WRAPPED_PUPable_decl_base_template(SINGLE_ARG(DampingFunction<VolumeDim, Fr>),
+                                     TimeDependentTripleGaussian);  // NOLINT
+
+  explicit TimeDependentTripleGaussian(CkMigrateMessage* /*unused*/) noexcept {}
```

Comment:
this should call the migrate constructor in the base class, and be moved to the `cpp` file. For checkpoint restart we will need the migration constructors to be functional